sourcegraph/cody
# fix: insert full code at cursor

close: https://github.com/sourcegraph/cody/issues/281

Fix issue where inserted code snippets removes `$`s

Cause: `editor.action.insertSnippet` would reformat the copied text that caused the inserted text to look different

```
await vscode.commands.executeCommand('editor.action.insertSnippet', {
                snippet: copiedText,
            })
```

Fix: Use WorkspaceEdit instead to replace current selection with the copied text

See Loom for fix: https://www.loom.com/share/7c238dfeda014e00b98b51aa1b5511ea?sid=ebca9d01-7f1b-4301-97b3-0451bbdc53ab

## Test Plan

Select code that has `$` in it, click on `explain code` recipe:
- copy the code should contain the correct code snippet with all `$` included
- insert the code at the cursor should contain the correct code snippet with all `$` included
